### PR TITLE
 ath79: add support for TP-Link Archer D50 V1

### PR DESCRIPTION
**Pull request by DavideFioravanti**
TP-Link Archer D50 v1 is a dual-band AC1200 router + modem.
The router section is based on Qualcomm/Atheros QCA9531 + QCA9882.
The "DSL" section is based on BCM6318 but it's currently not supported.

I tested only the initramfs image because I am still trying to
get the BCM6318 working. (WIP)

Router section - Specification:

CPU: QCA9531 650/600/200 MHz (CPU/DDR/AHB)
RAM: 64 MB (DDR2)
Flash: 8 MB (SPI NOR)
Wifi 2.4GHz: QCA95311 2T2R
Wifi 5GHz: QCA9982 2T2R
4x 10/100 Mbps Ethernet
8x LED, 3x button
UART header on PCB

Known issues:
DSL (eth0) (WIP)

Initramfs instruction under U-Boot, using UART:
tftpboot 0x81000000 openwrt-ath79-generic-tplink_archer-d50-v1-initramfs-kernel.bin
bootm 0x81000000

(**NOT TESTED**) Flash instruction under U-Boot, using UART:
tftp 0x81000000 openwrt-ar71xx-...-sysupgrade.bin
erase 0x9f020000 +$filesize
cp.b $fileaddr 0x9f020000 $filesize
reset

P.s. Probably we should swap the qca9531.dtsi and qca9533.dtsi file
P.p.s Although this is a Qualcomm device, I set TPLINK_FLASHLAYOUT := 8Mltq
because a "QCA layout" currently doesn't exists

Signed-off-by: Davide Fioravanti <pantanastyle@gmail.com>

### CODE REVIEW

**Comment by ynezz** [comment]
@DavideFioravanti It is WIP right? So it would be nice to mark it as such, so please change the topic to `WIP: ath79: add support for TP-Link Archer D50 V1` then please rebase it on the latest master and get rid of the `Merge...` commits, so from this state:

```
    Merge pull request #1 from openwrt/master
    Merge branch 'master' of https://github.com/openwrt/openwrt
    Merge branch 'master' of https://github.com/openwrt/openwrt
    Merge branch 'master' of https://github.com/openwrt/openwrt
    ath79: add support for TP-Link Archer D50 V1 
```

it's down to just this:

```
ath79: add support for TP-Link Archer D50 V1 
```

File: target/linux/ath79/dts/qca9531_tplink_archer-d50-v1.dts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot:	partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@20000 {
```
Comment (rmilecki) [inline target/linux/ath79/dts/qca9531_tplink_archer-d50-v1.dts]: Please add `compatible`, see
[OpenWrt-Devel] Specifying "firmware" partition format on all DTS targets
http://lists.infradead.org/pipermail/openwrt-devel/2018-November/014706.html
for details.

**Comment by ynezz** [comment]
Please either try to address the review comments or close this PR and reopen once you're ready. Thanks.

**Comment by ynezz** [comment]
Please re-open the PR once you find a time (or interest) to work on this.

**Pull request closed without merging**